src/IO/AssimpLoader/AssimpGeometryDataLoader.cpp

```diff
@@ -204,55 +204,54 @@ namespace Ra {
 
         void AssimpGeometryDataLoader::fetchVertices( const aiMesh& mesh, Asset::GeometryData& data ) {
             const uint size = mesh.mNumVertices;
-#if 0
-            std::vector< Core::Vector3 > vertex( size );
-            vertex.resize( size );
-            #pragma omp parallel for
-            for( uint i = 0; i < size; ++i ) {
-                vertex[i] = assimpToCore( mesh.mVertices[i] );
-            }
-#else
             auto &vertex = data.getVertices();
-            vertex.reserve(size/4);
+            vertex.reserve( size );
+            auto& duplicateTable = data.getDuplicateTable();
+            duplicateTable.reserve( size );
             std::map< Triplet, uint > uniqueTable;
-            std::map< uint, uint > duplicateTable;
             for( uint i = 0; i < size; ++i ) {
                 const Core::Vector3 v = assimpToCore( mesh.mVertices[i] );
                 const Triplet t( v );
                 auto it = uniqueTable.find( t );
-                if( ( it == uniqueTable.end() ) || data.isLoadingDuplicates() ) {
+                if( it == uniqueTable.end() || data.isLoadingDuplicates() ) {
+                    duplicateTable.push_back( vertex.size() );
+                    uniqueTable[t] = vertex.size();
                     vertex.push_back( v );
-                    uniqueTable[t]           = vertex.size() - 1;
-                    duplicateTable[i] = vertex.size() - 1;
                 } else {
-                    duplicateTable[i] = it->second;
+                    duplicateTable.push_back( it->second );
                 }
             }
-            data.setDuplicateTable( duplicateTable );
-#endif
+            vertex.shrink_to_fit();
+            duplicateTable.shrink_to_fit();
         }
 
         /// EDGE
         void AssimpGeometryDataLoader::fetchEdges( const aiMesh& mesh, Asset::GeometryData& data ) const {
             const uint size = mesh.mNumFaces;
-            auto &edge = data.getEdges(); edge.resize(size);
+            auto &edge = data.getEdges();
+            edge.resize( size );
             #pragma omp parallel for
-            for( int i = 0; i < int(size); ++i ) {
+            for( uint i = 0; i < size; ++i ) {
```

Comment:
Using unsigned loop counters require OpenMP 3.x.
https://stackoverflow.com/questions/2820621/why-arent-unsigned-openmp-index-variables-allowed

Since we use this multiple time in the code, please check in `cmake` that OpenMP version is 3.x or above.

Reply:
Fixed by 9cecd62.